### PR TITLE
Fix: Stop modifying DateTime  parameters during XML generation.

### PR DESCRIPTION
**Pull request by jodarove**
Pull request for https://github.com/braintree/braintree_php/issues/135.

### CODE REVIEW

File: lib/Braintree/Xml/Generator.php [hunk 1]
```diff
@@ -130,8 +130,10 @@ private static function _generateXmlAttribute($value)
      */
     private static function _dateTimeToXmlTimestamp($dateTime)
     {
-        $dateTime->setTimeZone(new DateTimeZone('UTC'));
-        return ($dateTime->format('Y-m-d\TH:i:s') . 'Z');
+        $_dateTime = clone $dateTime;
```
Comment (raymondberg) [inline lib/Braintree/Xml/Generator.php]: Style-wise, could this be `$dateTimeForUTC` or `$dateTimeWithUTC`?

File: tests/unit/Xml_GeneratorTest.php [hunk 2]
```diff
@@ -107,4 +107,28 @@ public function testEscapingSpecialChars()
         ]);
         $this->assertEquals($expected, $xml);
     }
+
+    public function testCanHandleDateTime()
```
Comment (raymondberg) [inline tests/unit/Xml_GeneratorTest.php]: I think a better test title would be `testDoesNotModifyDateTime`

**Comment by raymondberg** [comment]
@jodarove This looks great! I made two minor style comments. If you wouldn't mind amending your PR we could merge this in and get it out in an upcoming release.

**Comment by jodarove** [comment]
Thanks for your comments @raymondberg ! It sounds awesome!! I already amend the commit and updated the PR following your suggestions.

**Comment by pblesi** [comment]
@jodarove Thanks for the quick updates. I have merged these changes into our codebase and they will be included in our next release!

**Comment by raymondberg** [comment]
This went out with 3.14.0. Thanks again! 

https://github.com/braintree/braintree_php/commit/0c13b6235bc99c287d18e8cfc02359e66abaa3fb

**Pull request closed without merging**